src/librustc_mir/util/elaborate_drops.rs

```diff
@@ -752,7 +752,8 @@ impl<'l, 'b, 'tcx, D> DropCtxt<'l, 'b, 'tcx, D>
     fn open_drop<'a>(&mut self) -> BasicBlock {
         let ty = self.lvalue_ty(self.lvalue);
         match ty.sty {
-            ty::TyClosure(def_id, substs) => {
+            ty::TyClosure(def_id, substs) |
+            ty::TyGenerator(def_id, substs, _) => {
```

Comment:
This is not what `open_drop` is supposed to do - `open_drop` is supposed to drop *all* fields.While I don't think you can force an open drop for a generator ATM (because "normal people" can't access its fields), I'll prefer to at least have a safety assertion that this is only called when elaborating the drop of your own generator.